### ISSUE TITLE
numpy >=1.25.0 dependency in conda-forge?

### ISSUE THREAD

**Issue by plurch** [benfred/implicit#679]
I am trying to install `implicit` with `mamba` but it seems to have a dependency on `numpy >=1.25.0` in the `conda-forge` channel. This is causing an issue with my environment since `numba` does not support 1.25 yet (https://github.com/numba/numba/issues/8698). 

Is this expected? numpy 1.25.0 was released pretty recently (https://github.com/numpy/numpy/releases/tag/v1.25.0)

```
$ conda search implicit=0.7.0 --info -c conda-forge

implicit 0.7.0 py311hfbf440e_4
------------------------------
file name   : implicit-0.7.0-py311hfbf440e_4.conda
name        : implicit
version     : 0.7.0
build       : py311hfbf440e_4
build number: 4
size        : 452 KB
license     : MIT
subdir      : osx-64
url         : https://conda.anaconda.org/conda-forge/osx-64/implicit-0.7.0-py311hfbf440e_4.conda
md5         : f6f5aa6fa59ad01823d11f881b5871ad
timestamp   : 2023-06-22 16:52:49 UTC
constraints :
  - implicit-proc * cpu
dependencies:
  - libcxx >=15.0.7
  - numpy >=1.25.0,<2.0a0
  - python >=3.11,<3.12.0a0
  - python_abi 3.11.* *_cp311
  - scipy >=1.10.1,<2.0a0
  - tqdm >=4.27
```

**Comment by benfred** [comment]
I think those dependencies are probably too restrictive - it looks like scipy/numpy are set up with `pin_compatible` :

https://github.com/conda-forge/implicit-feedstock/blob/399286590ca22536ea555397c74c9960854c4ac0/recipe/meta.yaml#L67-L68

which afaict means that they get pinned to the version on the build machines.
https://docs.conda.io/projects/conda-build/en/latest/resources/variants.html#pinning-at-the-recipe-level

Implicit can successfully run with *much* older versions of both of these (for example, we work with scipy 0.16 which was released in 2015)

trying to fix this in https://github.com/conda-forge/implicit-feedstock/pull/83

**Comment by plurch** [comment]
Makes sense, looks like `pin_compatible` is not desired in this case.

Thanks for the follow-up - great library and blog by the way!

**Comment by benfred** [comment]
This should be fixed in the latest version up on conda-forge - which has removed the numpy pin_compatible.  Let me know if you still are facing any issues!

**Comment by plurch** [comment]
I tested installing `implicit` to my existing conda environment and also created a brand new environment without any issues. This is fixed with `implicit` version `0.7.1 ` - thanks! 👍
